MSW 초기 환경 설정

iyu88 opened the pull request:
<!-- 이 PR에서 구현한 작업들을 나열 -->

## 주요 작업사항

- MockUp API 를 작성하기 위해서 MSW 를 설정했습니다
  - src/mocks 폴더에서 확인할 수 있습니다. 
- eslint 관련 에러를 수정했습니다. 
  - airbnb 관련 dependency 가 설치되어 있지 않은 문제를 해결했습니다. 
  - 자세한 내용과 참고 링크는 commit message 에 있습니다.

<!-- (선택) 작업사항을 파악하기 쉽게 스크린샷 첨부 -->

## 스크린샷

![스크린샷 2023-01-09 22 12 41](https://user-images.githubusercontent.com/31645195/211324828-57acdb9f-ca14-49f1-8034-cc2a1810f25a.png)
![스크린샷 2023-01-09 22 12 48](https://user-images.githubusercontent.com/31645195/211324844-6f44300e-e7fe-42da-940d-095c5d02ac10.png)

```
import { useState, useEffect } from "react";

const Main = () => {
    const [ranking, setRanking] = useState([]);
    useEffect(() => {
        const fetchRanking = async () => {
            try {
                const data = await fetch("/ranking");
                const json = await data.json();
                setRanking(json);
            } catch (err) {
                console.log(err);
            } 
        }
        fetchRanking();
    }, []);

    return <div>
        {ranking.map(obj => JSON.stringify(obj))}
    </div>;
};
export default Main;

```
Main.tsx 에서 위 코드를 실행한 결과입니다. 정상적으로 Mock Data 를 불러옵니다. ( 현재 코드는 삭제했습니다. )

<!-- (선택) 리뷰요청사항 및 기타 하고싶은말 -->

## 팀원분들께

- 화이팅~~

iyu88 committed:
init: MSW 초기 환경 설정

MSW 로 목업 API 를 사용하기 위해서 초기 환경 설정을 진행합니다.
src/mocks 폴더 아래에 목업 API 생성과 관련된 파일을 정의했습니다.

iyu88 committed:
fix: ESlint 오류 수정

npm run start 를 했을 때 ESlint config 파일에서 적용했던 airbnb 룰이 적용되지 않았습니다.
해당 문제는 관련한 npm dependency 가 설치되지 않아서 발생했습니다.
airbnb 와 airbnb-typescript 와 관련한 의존성을 install-peerdep 를 사용하여 해결했습니다.
또한 dot-notation 이 ts 파일에서 적용되지 않는 오류가 있었습니다.
ESlint config 파일에서 parserOptions 를 수정하여 해결했습니다.

관련 링크
https://velog.io/@lumpenop/error-ESLint-Failed-to-load-config-airbnb-to-extend-from
https://github.com/airbnb/javascript/issues/2233
https://www.npmjs.com/package/eslint-config-airbnb-typescript
https://velog.io/@parkinhwa/%ED%94%84%EB%A1%9C%EC%A0%9D%ED%8A%B8-%ED%9A%8C%EA%B3%A0-%ED%86%A0%EC%9D%B4-%ED%94%84%EB%A1%9C%EC%A0%9D%ED%8A%B8-%ED%9A%8C%EA%B3%A0

caseBread commented on frontend/package.json:
```diff
@@ -51,9 +51,18 @@
     "@typescript-eslint/eslint-plugin": "^5.48.0",
     "@typescript-eslint/parser": "^5.48.0",
     "eslint": "^8.31.0",
+    "eslint-config-airbnb": "^19.0.4",
+    "eslint-config-airbnb-typescript": "^17.0.0",
```
맞네요.. 생각해보니까 airbnb룰 추가를 안했었네요 🫠

caseBread commented on frontend/src/mocks/handlers.ts:
```diff
@@ -0,0 +1,16 @@
+  rest.get('/ranking', (req, res, ctx) => {
+    return res(ctx.status(200), ctx.json(mock_ranking));
+  }),
+]
```
좋습니다!! 물론 이건 임시로 만든 방식이겠지만, 혹시 handler폴더 내에 각각의 핸들러 파일을 생성해보는 방식은 어떨지 건의드려봅니다

```
mocks
ㄴ handlers
  ㄴ ranking.ts
  ㄴ user.ts
  ㄴ index.ts // 모든 handler를 묶는 부분
```

이런 식으로요!

iyu88 replied:
말씀하신 방법이 더 깔끔한 것 같아서 변경했습니다!

caseBread reviewed (approved):
고생하셨습니다!!!

iyu88 committed:
refactor: handler 파일을 폴더로 변경

MSW 목업 API 를 정의했던 handler 파일을 폴더로 변경하고 각 API 를 파일로 관리합니다.

iyu88 committed:
fix: ESlint Error 해결을 위한 규칙 추가

mocks 에서 import 문을 사용할 때 발생한 에러를 no-extraneous-dependencies 규칙으로 해결합니다.
화살표 함수로 컴포넌트를 생성했을 때 발생한 에러를 function-component-definition 로 해결합니다.

관련 링크
https://stackoverflow.com/questions/44939304/eslint-should-be-listed-in-the-projects-dependencies-not-devdependencies
https://velog.io/@han1368/ESLint-%EC%A0%81%EC%9A%A9-%EC%8B%9C-%EC%83%9D%EA%B8%B4-%EC%9D%B4%EC%8A%88Function-component-is-not-a-function-declaration

caseBread commented:
LGTM!! 
merge는 pr올린사람이 하는걸루 할까요?
